### PR TITLE
Modify pex89000 device and add private data feature

### PR DESCRIPTION
**Pull request by MouchenHung-QUANTA**
Summary:
- Modify pex89000 common code
- Add *priv_data element into structure "sensor_cfg"

Test plan:
- Build code: pass

### CODE REVIEW

File: common/dev/pex89000.c [hunk 1]
```diff
@@ -131,231 +118,375 @@ static bool pex89000_chime_read(uint8_t bus, uint8_t addr, uint32_t oft, uint8_t
...
+				goto exit;
+			}
+
+			if (pex89000_chime_to_axi_read(bus, addr, 0xFFE00010, &resp)) {
+				printf("CHIME to AXI Write 0xFFE00010 fail!\n");
+				goto exit;
+			}
+
+			temp = (float)(366.812 - 0.23751 * (float)(resp & 0x7FF));
+			temp_arr[i] = temp;
```
Comment (GoldenBug) [inline common/dev/pex89000.c:382]: Can you please add comments describing what all these addresses are for?
Possibly add `#define` for the various constant values used here.

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: common/dev/pex89000.c [hunk 2]
```diff
@@ -1,125 +1,112 @@
...
  * buf: encoded byte array to send to pesw
  */
-
-static void pex89000_i2c_encode(uint32_t oft, uint8_t be, uint8_t cmd, uint8_t access_mode,
-				HW_I2C_Cmd *buf)
+static void pex89000_i2c_encode(uint32_t oft, uint8_t be, uint8_t cmd, HW_I2C_Cmd *buf)
 {
-	if (buf == NULL) {
+	if (!buf) {
+		printf("%s: *buf does not exist!\n", __func__);
```
[the remark this answers is not in the document]
Comment (MouchenHung-QUANTA) [reply common/dev/pex89000.c:87]: Will be modified in next commit, thanks.

File: common/dev/pex89000.c [hunk 3]
```diff
@@ -131,231 +118,375 @@ static bool pex89000_chime_read(uint8_t bus, uint8_t addr, uint32_t oft, uint8_t
...
+				    uint8_t data_len)
 {
-	if (data == NULL) {
-		return false;
-	}
-
-	if (!set_full_addr(bus, addr, oft)) {
-		return false;
+	if (!data) {
+		printf("%s: *data does not exist!\n", __func__);
```
[the remark this answers is not in the document]
Comment (MouchenHung-QUANTA) [reply common/dev/pex89000.c:134]: Will be modified in next commit, thanks.

File: common/dev/pex89000.c [hunk 4]
```diff
@@ -131,231 +118,375 @@ static bool pex89000_chime_read(uint8_t bus, uint8_t addr, uint32_t oft, uint8_t
...
+	return rc;
 }
 
-bool pex89000_die_temp(uint8_t bus, uint8_t addr, sensor_val *val)
+uint8_t pex_access_engine(uint8_t bus, uint8_t addr, uint8_t idx, pex_access_t key, uint32_t *resp)
 {
-	if (val == NULL) {
-		return false;
+	if (!resp) {
+		printf("%s: *resp does not exist!\n", __func__);
```
[the remark this answers is not in the document]
Comment (MouchenHung-QUANTA) [reply common/dev/pex89000.c:249]: Will be modified in next commit, thanks.

File: common/dev/pex89000.c [hunk 5]
```diff
@@ -131,231 +118,375 @@ static bool pex89000_chime_read(uint8_t bus, uint8_t addr, uint32_t oft, uint8_t
...
+
+	return rc;
+}
+
+static uint8_t pex89000_temp(uint8_t bus, uint8_t addr, pex_dev_t dev, uint32_t *val)
+{
+	uint8_t rc = pex_api_unspecific_err;
+
+	if (!val) {
+		printf("%s: *val does not exist !\n", __func__);
```
[the remark this answers is not in the document]
Comment (MouchenHung-QUANTA) [reply common/dev/pex89000.c:325]: Will be modified in next commit, thanks.

File: common/dev/pex89000.c (hunk 1 above)
Comment (MouchenHung-QUANTA) [reply common/dev/pex89000.c:382]: Due to new PEX89000 chip is on development stage, some registers or data are not in current spec. Instead, those registers came from Broadcom debug tool. 
Dummy macro would be taken temporarily, thanks.

**Pull request closed without merging**